apps/site/src/pages/account/api-keys-panel/api-keys-list.tsx

```diff
@@ -0,0 +1,200 @@
+import { useApiKeys } from "./api-keys-context";
+import { ApiKeyCard } from "./api-keys-list/api-key-card";
+import { ApiKeyTableRow } from "./api-keys-list/api-key-table-row";
+import { MobileApiKeyItem } from "./api-keys-list/mobile-api-key-item";
+import { RevokeApiKeyCard } from "./api-keys-list/revoke-api-key-card";
+import { ApiKeyItemProps, ApiKeyProps } from "./types";
+
+export const ApiKeysList = () => {
+  const {
+    apiKeys,
+    keyActionStatus,
+    setKeyActionStatus,
+    newlyCreatedKeyIds,
+    isCreatingNewKey,
+    setApiKeys,
+    setIsCreatingNewKey,
+    setNewlyCreatedKeyIds,
+    fetchAndSetApiKeys,
+  } = useApiKeys();
+
+  const revokeApiKey = async (publicId: string) => {
+    const res = await apiClient.revokeApiKey({ publicId });
+
+    if (res.error) {
+      throw new Error(res.error.message);
+    }
+
+    setKeyActionStatus(undefined);
+    setApiKeys(apiKeys.filter((key) => key.publicId !== publicId));
+  };
+
+  const renameApiKey = async (publicId: string, displayName: string) => {
+    await apiClient.updateApiKey({ publicId, displayName });
+    setKeyActionStatus(undefined);
+    setApiKeys(
+      apiKeys.map((key) =>
+        key.publicId === publicId ? { ...key, displayName } : key,
+      ),
+    );
+  };
+
+  const createKey = async (displayName: string) => {
+    const { data } = await apiClient.generateApiKey({ displayName });
+
+    if (data) {
+      await fetchAndSetApiKeys();
+      setNewlyCreatedKeyIds((ids) => [...ids, data.apiKey]);
+      setIsCreatingNewKey(false);
+    }
+  };
+
+  const createKeyCard = (
+    <ApiKeyCard
+      onClose={() => setIsCreatingNewKey(false)}
+      onSubmit={createKey}
+      submitTitle="Create key"
+      inputLabel="Name your new key"
+    />
+  );
+
+  const generateApiKeyItemProps = (data: ApiKeyProps): ApiKeyItemProps => {
+    const dismissKeyAction = () => setKeyActionStatus(undefined);
+
+    const { displayName, publicId } = data;
+    return {
+      apiKey: data,
+      fullKeyValue: newlyCreatedKeyIds.find((key) => key.includes(publicId)),
+      keyAction:
+        keyActionStatus?.publicId === publicId
+          ? keyActionStatus.action
+          : undefined,
+      renameApiKeyCard: (
+        <ApiKeyCard
+          onClose={dismissKeyAction}
+          defaultValue={displayName}
+          showDiscardButton
+          submitTitle="Rename key"
+          inputLabel="Rename your key"
+          onSubmit={async (newDisplayName) =>
+            renameApiKey(publicId, newDisplayName)
+          }
+        />
+      ),
+      revokeApiKeyCard: (
+        <RevokeApiKeyCard
+          onClose={dismissKeyAction}
+          displayName={displayName}
+          onRevoke={async () => revokeApiKey(publicId)}
+        />
+      ),
+    };
+  };
+
+  return (
+    <>
+      <Box
+        sx={(theme) => ({
+          [theme.breakpoints.up("md")]: { display: "none" },
+        })}
+      >
+        {apiKeys.map((data, index) => (
+          <>
```

Comment:
This fragment is missing a key, and the elements within it don't need a key.

Reply:
Fixed here [fix no key warning](https://github.com/blockprotocol/blockprotocol/pull/1257/commits/ee0a21b2939d2963a9fa4ea524dc15bedb923917)